UserService/Exceptions/ExceptionHandlerMiddleware.cs

```diff
@@ -0,0 +1,47 @@
+using System.Net;
+
+namespace UserService.Exceptions {
+    public class ExceptionHandlerMiddleware
```

Comment:
Can you please explain the logic behind this class

Reply:
this class will serve us to centralize exception handling in the service, instead of handling exceptions in each endpoint separately